### PR TITLE
python pip cache to be  located at /usr/local/share/pip-cache

### PR DESCRIPTION
**Pull request by arroyc**
python package cache to be located at /usr/local/share/pip-cache

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -1,6 +1,11 @@
 declare -r TS_FMT='[%T%z] '
 declare -r REQS_NOT_FOUND_MSG='Could not find requirements.txt; Not running pip install'
 echo "Python Version: $python"
+PIP_CACHE_DIR=/usr/local/share/pip-cache
```
Comment (kichalla) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: Curious, why this particular location and not something like let's say `/tmp/pip-cache`?

Comment (kichalla) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: Can you kick off a Nightly build against this change? I am curious if this stuff works with `oryxtests/build` image? You probably already tried it?

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 2]
```diff
@@ -34,7 +39,7 @@ then
 	echo "Done in $ELAPSED_TIME sec(s)."
 
 	echo "Running pip install..."
-	pip install --prefer-binary -r requirements.txt | ts $TS_FMT
+	pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r requirements.txt | ts $TS_FMT
```
Comment (kichalla) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: How is the performance of a second build? do you see a similar experience as before this change?

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (arroyc) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: sure .. I know what you are trying to get .. but I added permission as we have for yarn cache, but yet. nightly-build is a good idea. Answering your first question, I tried to keep the location/path similar to the yarn-cache

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 2 above)
Comment (arroyc) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: locally I didn't see any difference (i.e. running integration tests etc on local containers)

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (kichalla) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: Cool

**Pull request merged**